From 91eda209ba094c859befbe379805eac57bddd123 Mon Sep 17 00:00:00 2001
From: Alaa Hleihel <ahleihel@redhat.com>
Date: Sun, 10 May 2020 14:52:22 -0400
Subject: [PATCH 049/312] [netdrv] net/mlx5: Kconfig: Fix MLX5_CORE dependency
 with PCI_HYPERV_INTERFACE

Message-id: <20200510145245.10054-60-ahleihel@redhat.com>
Patchwork-id: 306600
Patchwork-instance: patchwork
O-Subject: [RHEL8.3 BZ 1789378 v2 59/82] net/mlx5: Kconfig: Fix MLX5_CORE dependency with PCI_HYPERV_INTERFACE
Bugzilla: 1789378
RH-Acked-by: Kamal Heib <kheib@redhat.com>
RH-Acked-by: Jarod Wilson <jarod@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Jonathan Toppins <jtoppins@redhat.com>

Bugzilla: http://bugzilla.redhat.com/1789378
Upstream: v5.4-rc1

commit 4057a7652b74af25ba1197689fc144cdb766f423
Author: Mao Wenan <maowenan@huawei.com>
Date:   Tue Aug 27 11:12:51 2019 +0800

    net/mlx5: Kconfig: Fix MLX5_CORE dependency with PCI_HYPERV_INTERFACE

    When MLX5_CORE=y and PCI_HYPERV_INTERFACE=m, below errors are found:
    drivers/net/ethernet/mellanox/mlx5/core/en_main.o: In function `mlx5e_nic_enable':
    en_main.c:(.text+0xb649): undefined reference to `mlx5e_hv_vhca_stats_create'
    drivers/net/ethernet/mellanox/mlx5/core/en_main.o: In function `mlx5e_nic_disable':
    en_main.c:(.text+0xb8c4): undefined reference to `mlx5e_hv_vhca_stats_destroy'

    Fix this by making MLX5_CORE imply PCI_HYPERV_INTERFACE.

    Fixes: cef35af34d6d ("net/mlx5e: Add mlx5e HV VHCA stats agent")
    Signed-off-by: Mao Wenan <maowenan@huawei.com>
    Signed-off-by: Saeed Mahameed <saeedm@mellanox.com>

Signed-off-by: Alaa Hleihel <ahleihel@redhat.com>
Signed-off-by: Frantisek Hrbata <fhrbata@redhat.com>
---
 drivers/net/ethernet/mellanox/mlx5/core/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/ethernet/mellanox/mlx5/core/Kconfig b/drivers/net/ethernet/mellanox/mlx5/core/Kconfig
index 92a561176705..ae7c28ba9f5a 100644
--- a/drivers/net/ethernet/mellanox/mlx5/core/Kconfig
+++ b/drivers/net/ethernet/mellanox/mlx5/core/Kconfig
@@ -9,6 +9,7 @@ config MLX5_CORE
 	imply PTP_1588_CLOCK
 	imply VXLAN
 	imply MLXFW
+	imply PCI_HYPERV_INTERFACE
 	default n
 	---help---
 	  Core driver for low level functionality of the ConnectX-4 and
-- 
2.13.6

